ios/FluentUI/Core/Theme/Tokens/FontInfo.swift

```diff
@@ -76,22 +76,44 @@ public extension Font {
 
 extension UIFont {
     @objc public static func fluent(_ fontInfo: FontInfo, shouldScale: Bool = true) -> UIFont {
-        let unscaledFont: UIFont
+        return fluent(fontInfo, shouldScale: shouldScale, contentSizeCategory: nil)
+    }
+
+    static func fluent(_ fontInfo: FontInfo, shouldScale: Bool = true, contentSizeCategory: UIContentSizeCategory?) -> UIFont {
+        let traitCollection: UITraitCollection?
+        if let contentSizeCategory = contentSizeCategory {
+            traitCollection = .init(preferredContentSizeCategory: contentSizeCategory)
+        } else {
+            traitCollection = nil
+        }
+
+        let weight = uiWeight(fontInfo.weight)
 
         if let name = fontInfo.name,
            let font = UIFont(name: name, size: fontInfo.size) {
             // Named font
-            unscaledFont = font.withWeight(uiWeight(fontInfo.weight))
+            let unscaledFont = font.withWeight(weight)
+            if shouldScale {
+                let fontMetrics = UIFontMetrics(forTextStyle: uiTextStyle(fontInfo.textStyle))
+                return fontMetrics.scaledFont(for: unscaledFont, compatibleWith: traitCollection)
+            } else {
+                return unscaledFont
+            }
         } else {
             // System font
-            unscaledFont = .systemFont(ofSize: fontInfo.size, weight: uiWeight(fontInfo.weight))
-        }
+            if !shouldScale {
+                return .systemFont(ofSize: fontInfo.size, weight: weight)
+            }
 
-        if shouldScale {
-            let fontMetrics = UIFontMetrics(forTextStyle: uiTextStyle(fontInfo.textStyle))
-            return fontMetrics.scaledFont(for: unscaledFont)
-        } else {
-            return unscaledFont
+            let textStyle = uiTextStyle(fontInfo.textStyle)
+            if FontInfo.sizeTuples.contains(where: { $0.size == fontInfo.size }) {
+                // System-recognized font size, let the OS scale it for us
+                return UIFont.preferredFont(forTextStyle: textStyle, compatibleWith: traitCollection).withWeight(weight)
+            }
+
+            // Custom font size, we need to scale it ourselves
+            let fontMetrics = UIFontMetrics(forTextStyle: textStyle)
+            return fontMetrics.scaledFont(for: .systemFont(ofSize: fontInfo.size, weight: weight), compatibleWith: traitCollection)
```

Comment:
This seems like it'll be much more inconsistent - using `UIFont.preferredFont` for system fonts sized 22.0, but `fontMetrics.scaledFont` if it's 21.0 or 23.0. If we really do see significantly different behavior between these two APIs, won't that appear as a noticeable misalignment?

Reply:
Valid point.

The problem ultimately lies in the fact that `UIFont.preferredFont` and `fontMetrics.scaledFont` give different values for font sizes aligned with built-in styles. I think the solution lies in what kinds of typography overrides we want to encourage or discourage clients from using.

Reply:
As discussed offline, this is something we can live with for now.